api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -88,45 +99,86 @@ public Router initializeRouter(Vertx vertx) {
+
+    // should this be admin only?
+    Route getUsersRoute = router.route("/protected/users");
+    getUsersRoute.handler(this::handleGetAllUsers);
+
+    // Get User : Protected POST method
+    Route getUserRoute = router.route("/protected/user/:id");
+    getUserRoute.handler(this::handleGetUser);
+
+    Route updateUserRoute = router.put("/protected/users");
+    updateUserRoute.handler(this::handleUpdateUser);
+
+    // what happens when a user deletes their account, its fine, but what about when
+    // an admin does it?
+    Route deleteUserRoute = router.delete("/protected/user");
+    deleteUserRoute.handler(this::handleDeleteUser);
+
+    // should logout be a POST or GET? we dont supply information but it is
+    // modifying a resource...
+    // 200 or 201
     Route logoutRoute = router.route("/logout");
     logoutRoute.handler(this::handleLogout);
 
-    /*
-     * Session counter is no longer needed as we dont use sessions. Route
-     * sessionRoute = router.route("/session");
-     * sessionRoute.handler(this::handleSession);
-     */
+    // Route sessionRoute = router.route("/session");
+    // sessionRoute.handler(this::handleSession);
+
+    Route getEventsRoute = router.route().path("/protected/events");
+    getEventsRoute.handler(this::handleGetEvents);
+
+    Route createEventRoute = router.post("/admin/event");
+    createEventRoute.handler(this::handleCreateEvent);
+
+    Route getEventRoute = router.get("/protected/event/:id");
+    getEventRoute.handler(this::handleGetEvent);
 
-    Route getMemberRoute = router.route().path("/api/v1/members");
-    getMemberRoute.handler(this::handleGetMemberRoute);
+    Route updateEventRoute = router.put("/admin/event/:id");
+    updateEventRoute.handler(this::handleUpdateEvent);
 
-    Route createMeetingRoute = router.route("/protected/createmeeting");
-    createMeetingRoute.handler(this::handleCreateMeeting);
+    Route deleteEventRoute = router.delete("/admin/event/:id");
+    deleteEventRoute.handler(this::handleDeleteEvent);
 
-    Route attendMeetingRoute = router.route("/protected/attendmeeting");
-    attendMeetingRoute.handler(this::handleAttendMeeting);
+    // asd asdsad
+
+    Route attendEventRoute = router.post("/protected/eventcheckin/:id");
+    attendEventRoute.handler(this::handleAttendEvent);
+
+    Route getEventUsersRoute = router.get("/protected/eventcheckin/:id");
+    getEventUsersRoute.handler(this::handleGetEventUsers);
 
     return router;
   }
 
   /**
-   * Add a handler for getting all members.
+   * Add a handler for getting all users.
    */
-  private void handleGetMemberRoute(RoutingContext ctx) {
+  private void handleGetAllUsers(RoutingContext ctx) {
     HttpServerResponse response = ctx.response();
     response.putHeader("content-type", "application/json");
-    List<MemberReturn> members = processor.getAllMembers();
+    List<UserReturn> users = processor.getAllUsers();
 
-    String memberJson = null;
+    String userJson = null;
```

Comment:
Put this and the end() call in the try and if you catch an exception then return a standard error message to the client